src/span.h

```diff
@@ -11,49 +11,49 @@
 #include <type_traits>
 #include <utility>
 
-/** A Span is an object that can refer to a contiguous sequence of objects.
+/** A std::span is an object that can refer to a contiguous sequence of objects.
```

Comment:
shouldn't these go to the [developer notes](https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md?plain=1#L859-L863) now? We don't have files for other std types locally which explain their behavior.

Reply:
I find the suggestions helpful, I'm fine with moving them to doc separately.
The `span's` should likely be adjusted here as well, maybe to `instances of std::span` or something similar.